lib/graphql/query/serial_execution/value_resolution.rb

```diff
@@ -7,7 +7,8 @@ def self.resolve(parent_type, field_defn, field_type, value, selection, query_ct
           case value
           when GraphQL::ExecutionError, NilClass
             if field_type.kind.non_null?
-              query_ctx.schema.type_error(value, field_defn, parent_type, query_ctx)
+              type_error = GraphQL::InvalidNullError.new(parent_type, field_defn, value)
+              query_ctx.schema.type_error(type_error, query_ctx)
```

Comment:
This is where you shouldn't be calling type_error if the value is an GraphQL::ExecutionError.